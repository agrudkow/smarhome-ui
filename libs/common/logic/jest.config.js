module.exports = {
  name: 'common-logic',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/common/logic',
};
