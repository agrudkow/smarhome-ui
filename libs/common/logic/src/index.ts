export * from './lib/use-accordion';
export * from './lib/use-window-dimensions';
