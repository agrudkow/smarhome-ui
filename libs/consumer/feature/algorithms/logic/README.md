# consumer-feature-algorithms-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-algorithms-logic` to execute the unit tests via [Jest](https://jestjs.io).
