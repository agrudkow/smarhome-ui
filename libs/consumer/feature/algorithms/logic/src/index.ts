export * from './lib/algorithm.interface';
export * from './lib/algorithms-cells-parser';
export * from './lib/algorithms-data-parser';
