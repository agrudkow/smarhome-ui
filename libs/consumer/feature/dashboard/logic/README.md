# consumer-feature-dashboard-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-dashboard-logic` to execute the unit tests via [Jest](https://jestjs.io).
