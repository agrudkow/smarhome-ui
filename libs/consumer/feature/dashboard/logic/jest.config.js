module.exports = {
  name: 'consumer-feature-dashboard-logic',
  preset: '../../../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory:
    '../../../../../coverage/libs/consumer/feature/dashboard/logic',
};
