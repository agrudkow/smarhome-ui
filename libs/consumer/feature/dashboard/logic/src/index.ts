export * from './lib/consumer-feature-dashboard-logic';
