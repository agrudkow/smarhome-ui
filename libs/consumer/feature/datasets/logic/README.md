# consumer-feature-datasets-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-datasets-logic` to execute the unit tests via [Jest](https://jestjs.io).
