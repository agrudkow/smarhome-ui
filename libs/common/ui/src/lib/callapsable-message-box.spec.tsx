import React from 'react';
import { customRender } from '@smarthome/common/logic';
import CallapsableMessageBox from './callapsable-message-box';

describe(' CallapsableMessageBox', () => {
  it('should render successfully', () => {
    const { baseElement } = customRender(<CallapsableMessageBox show={true} />);
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { baseElement } = customRender(<CallapsableMessageBox show={true} />);
    expect(baseElement.firstChild).toMatchSnapshot();
  });

  it('should be hidden on show set to false', () => {
    const { container } = customRender(<CallapsableMessageBox show={false} />);
    expect(container.firstChild).toHaveStyleRule('visibility', 'hidden');
    expect(container.firstChild).toHaveStyleRule('height', '0');
    expect(container.firstChild).toHaveStyleRule('margin', '0');
  });
});
