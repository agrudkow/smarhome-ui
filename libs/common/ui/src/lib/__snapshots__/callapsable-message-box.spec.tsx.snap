// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` CallapsableMessageBox should match snapshot 1`] = `
.c0 {
  color: #555;
  font-size: 14px;
  line-height: 20px;
  -webkit-letter-spacing: -0.3px;
  -moz-letter-spacing: -0.3px;
  -ms-letter-spacing: -0.3px;
  letter-spacing: -0.3px;
  padding: 5px;
  width: calc(100% - 14px);
  height: 0px;
  margin: 10px 0;
  visibility: visible;
  overflow: hidden;
  -webkit-transition: height 0.3s ease-in-out,visibility 0.3s ease-in-out, margin 0.3s ease-in-out;
  transition: height 0.3s ease-in-out,visibility 0.3s ease-in-out, margin 0.3s ease-in-out;
  border: 2px dashed #1BBDCE;
  border-radius: 6px;
  text-align: justify;
}

@media (min-width:769px) {
  .c0 {
    font-size: 14px;
    line-height: 20px;
  }
}

@media (min-width:1024px) {
  .c0 {
    font-size: 16px;
    line-height: 24px;
  }
}

@media (max-width:480px) {
  .c0 {
    margin: 5px 0;
  }
}

<div>
  <div
    class="c0"
  >
    <div />
  </div>
</div>
`;
